code-studio/build-commands.js

```diff
@@ -0,0 +1,114 @@
+  return [
+    'browserify --debug ' +  browserifyInputFiles,
+    "-p [ factor-bundle -o '> " + buildPath + "`basename $FILE .js`.js' ]",
+    '-o ' + buildPath + 'code-studio-common.js'
+  ].join(" \\\n    ");
+};
+
+/**
+ * Generate command to:
+ * Creates given director(y|ies) to any depth if it does not exist, no-op if
+ * it does already exist.
+ * @param {!string} dir
+ * @returns {string}
+ */
+exports.ensureDirectoryExists = function (dir) {
+  return 'mkdir -p ' + dir;
+};
+
+/**
+ * Execute a sequence of shell commands as a build script, with useful output
+ * wherever possible.
+ * Will call process.exit() if the command fails for any reason.
+ * @param {string[]} commands - array of shell commands to be executed in sequence.
+ */
+exports.execute = function (commands) {
+  // Run build
+  var buildCommands = commands.join(" && \\\n");
+  console.log(buildCommands);
+
+  try {
+    // For documentation on synchronous execution of shell commands from node scripts, see:
+    // https://nodejs.org/docs/latest/api/child_process.html#child_process_synchronous_process_creation
+    var result = child_process.execSync(buildCommands, {
+      env: _.extend({}, process.env, {
+        PATH: './node_modules/.bin:' + process.env.PATH
+      }),
+      stdio: 'inherit'
+    });
+  } catch (e) {
+    process.exit(e.status);
+  }
+};
+
+/**
+ * Generate command to:
+ * Process scss files into css files using node-sass.
+ * @param {string} srcPath - Path to root of SCSS source files, absolute
+ *        or relative to execution path for this script (which is the code-studio
+ *        folder for this build system), with trailing slash.
+ * @param {string} buildPath - Path to root of output directory, absolute or
+ *        relative to execution path for this script (which is the code-studio
+ *        folder for this build system), with trailing slash.
+ * @param {string[]} files - List of files to build, given as paths rooted at
+ *        the srcPath given.  Each will map to an output file.
+ * @param {string[]} includePaths - List of paths to search for files included
+ *        via scss import directives, rooted at the working directory, with NO
+ *        trailing slash.
+ * @param {boolean} [shouldMinify] if provided and TRUE, will build minified
+ *        output files (with .min.css extensions) instead of unminified output.
+ * @returns {string}
+ */
+exports.sassCommand = function (srcPath, buildPath, files, includePaths, shouldMinify) {
+  var includePathArgs = includePaths.map(function (path) {
+    return '--include-path ' + path;
+  }).join(' ');
+
+  return files.map(function (file) {
+    if (shouldMinify) {
+      return 'node-sass --output-style compressed ' + includePathArgs + ' ' +
+          srcPath + file + ' ' +
+          buildPath + path.basename(file, '.scss') + '.min.css';
+    }
+
+    return 'node-sass ' + includePathArgs + ' ' +
+        srcPath + file + ' ' +
+        buildPath + path.basename(file, '.scss') + '.css';
```

Comment:
I stuck with the pattern from the browserify command; in that case, the two paths were different enough that I decided it was more readable to separate them.  Here it's probably overkill.  I'll go ahead and take your suggestion.

On a related note, we're not actually using the build:min options anywhere yet - right now the Rails Asset Pipeline does our minification for prod, not this process.  If we're fine with that we could remove all the minifcation code from here.  I've just been keeping it working as a proof-of-concept.